tests/integration/test_store.py

```diff
@@ -499,6 +499,81 @@ def transform(e):
             assert event["logentry"]["formatted"] == f"otherkey{i}"
 
 
+def test_rate_limit_metric_bucket(
+    mini_sentry,
+    relay_with_processing,
+    metrics_consumer,
+):
+    metrics_consumer = metrics_consumer()
+
+    bucket_interval = 1  # second
+    relay = relay_with_processing(
+        {
+            "processing": {"max_rate_limit": 2 * 86400},
+            "aggregator": {
+                "bucket_interval": bucket_interval,
+                "initial_delay": 0,
+                "debounce_delay": 0,
+            },
+        }
+    )
+
+    metric_bucket_limit = 5
+    buckets_sent = 10
+
+    project_id = 42
+    projectconfig = mini_sentry.add_full_project_config(project_id)
+    mini_sentry.add_dsn_key_to_project(project_id)
+
+    public_keys = mini_sentry.get_dsn_public_key_configs(project_id)
+    key_id = public_keys[0]["numericId"]
+    projectconfig["config"]["quotas"] = [
+        {
+            "id": f"test_rate_limiting_{uuid.uuid4().hex}",
+            "scope": "key",
+            "scopeId": str(key_id),
+            "categories": ["metric_bucket"],
+            "limit": metric_bucket_limit,
+            "window": 86400,
+            "reasonCode": "throughput rate limiting",
+        }
+    ]
+
+    def generate_ticks():
+        # Generate a new timestamp for every bucket, so they do not get merged by the aggregator
```

Comment:
We can also use tags for this, should be easier than having timestamps tied to `bucket_interval`